src/crimson/os/seastore/segment_cleaner.cc

```diff
@@ -449,8 +449,11 @@ SegmentCleaner::init_segments_ret SegmentCleaner::init_segments() {
 	      "ExtentReader::init_segments: journal segment {}",
 	      segment_id);
 	    segment_set.emplace_back(std::make_pair(segment_id, std::move(header)));
+	    init_mark_segment_closed(
```

Comment:
Seems we can change this lambda block to something like:

```
if (!header.out_of_line) {
  segment_set.emplace_back(std::make_pair(segment_id, std::move(header)));
}
init_mark_segment_closed(
  segment_id,
  header.journal_segment_seq,
  false);
```

Reply:
LGTM otherwise

Reply:
Agreed, I plan to improve this when rebase the pending commit https://github.com/ceph/ceph/pull/44809/commits/159d5b06c52235afe92275ba1c0abd501aec2cbe onto this PR.

That commit enables us to print the segment type directly, and no need to pass the 3rd parameter about whether the segment is ool. So we can consolidate both loggers and `init_mark_segment_closed()` easier.